sfs/plot.py

```diff
@@ -281,6 +281,9 @@ def soundfield(p, grid, xnorm=None, cmap='coolwarm_clip', vmin=-2.0, vmax=2.0,
     elif plotting_plane == 'yz':
         x, y = grid[[1, 2]]
 
+    dx = np.asscalar(np.diff(x[0, :2])) / 2
+    dy = np.asscalar(np.diff(y[:2, 0])) / 2
```

Comment:
This looks like it makes more assumptions than it should.

Does this work when using a vertical grid slice?

For example using `sfs.util.xyz_grid()` with `x=0` or `y=0` instead of `z=0`?

Reply:
Yes, it was tested for `xz` plane and `yz` plane.

![soundfield_xz](https://user-images.githubusercontent.com/8125968/53596169-70cf8d80-3b9f-11e9-9971-2e5f1ea62f17.png)
![soundfield_yz](https://user-images.githubusercontent.com/8125968/53596176-73ca7e00-3b9f-11e9-8a2e-2f8bafaf1b52.png)

In the following part of the code,
_nonscalar_ axes are defined as `x` and `y`,
which are always column and row vectors, respectively.

https://github.com/sfstoolbox/sfs-python/blob/c10fa334eb832f707078a2bd626e9746c1607b84/sfs/plot.py#L277-L282

This will work only for equidistant girds, though.
If we want to plot a sound field on a nonuniform gird,
we should figure out something else.
BTW, I don't know if `imshow` or `pcolormesh` supports non-uniform grids,

Reply:
Thanks for testing this!

> which are always column and row vectors, respectively.

Yes, but only when using `sfs.util.xyz_grid()`.

However, it doesn't work when using `numpy.meshgrid()`, for example:

```python
grid = np.meshgrid(
    [-2, -1.5, -1, -0.5, 0, 0.5, 1, 1.5, 2],
    [-2, -1.5, -1, -0.5, 0, 0.5, 1, 1.5, 2],
    [0],
    #sparse=True,
)
```

... which without your change works (with and without `sparse`) and should result in the same plot.

I think a more robust (and shorter, and easier-to-understand) solution would be something like this:

```python
dx = 0.5 * x.ptp() / p.shape[1]
dy = 0.5 * y.ptp() / p.shape[0]
```

This doesn't make any assumptions about the shape of `x` and `y` (except that they must not be empty).

> If we want to plot a sound field on a nonuniform gird,
> we should figure out something else.

I don't think that `imshow()` does the right thing on nonuniform grids. It has to assume uniformity.

I think we should make calculations work with non-uniform grids, but I think it's fine to assume uniform grids for this kind of plotting functions.
We could at some point add plotting functions for point clouds similar to http://matlab.sfstoolbox.org/en/2.4.3/custom-grid/, but those should be separate functions.

BTW, calculating those point clouds should already work (but I haven't thoroughly tested this).

Reply:
We should probably add the uniformity assumption to the docs?

Reply:
> ```python
> dx = 0.5 * x.ptp() / p.shape[1]
> dy = 0.5 * y.ptp() / p.shape[0]
> ```

Yeah, that's much better. Thanks.
I will take this over.

> BTW, calculating those point clouds should already work (but I haven't thoroughly tested this).

The data should then be plotted using `scatter` I guess?

Reply:
> We should probably add the uniformity assumption to the docs?

Would that be necessary?
`imshow()` isn't suited for nonuniform grid anyway.

Reply:
> Would that be necessary?

I don't know.
Probably not. But it wouldn't hurt either ...